### ISSUE TITLE
New frontend build systems leads to frequent failures in the build

### ISSUE THREAD

**Issue by drexin** [mesosphere/marathon#1017]
This is the error:

```
/npm WARN optional dep failed, continuing fsevents@0.3.1
\npm ERR! cb() never called!
npm ERR! not ok code 0

The command "pushd src/main/resources/assets && npm install -g broccoli-cli && npm install && npm run dist && popd" failed and exited with 1 during .
```

**Comment by mlunoe** [comment]
Thank you. Will look at it asap!

**Comment by ConnorDoyle** [comment]
Saw this again, here is a more complete log:

```
Using worker: worker-linux-7-2.bb.travis-ci.org:travis-linux-1
system_info
Build system information
Build language: scala
Build image provisioning date and time
Sun Dec  7 05:49:51 UTC 2014
lsb_release -a
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04 LTS
Release:	12.04
Codename:	precise
Cookbooks Version
5736160 https://github.com/travis-ci/travis-cookbooks/tree/5736160
GCC version
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
LLVM version
clang version 3.4 (tags/RELEASE_34/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
Pre-installed Ruby versions
ruby-1.9.3-p551
Pre-installed Node.js versions
v0.10.33
Pre-installed Go versions
go1.3.3
Redis version
redis-server 2.8.18
riak version
1.4.12
MongoDB version
MongoDB 2.4.12
CouchDB version
couchdb 1.6.1
Neo4j version
1.9.4
Cassandra version
2.0.9
RabbitMQ Version
3.4.2
ElasticSearch version
1.4.0
Installed Sphinx versions
2.0.10
2.1.9
2.2.4
Default Sphinx version
2.1.9
Installed Firefox version
firefox 31.0esr
PhantomJS version
1.9.8
ant -version
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
mvn -version
Apache Maven 3.2.3 (33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4; 2014-08-11T20:58:10+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_72, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "2.6.32-042stab090.5", arch: "amd64", family: "unix"
gradle -version
------------------------------------------------------------
Gradle 2.0
------------------------------------------------------------
Build time:   2014-07-01 07:45:34 UTC
Build number: none
Revision:     b6ead6fa452dfdadec484059191eb641d817226c
Groovy:       2.3.3
Ant:          Apache Ant(TM) version 1.9.3 compiled on December 23 2013
JVM:          1.7.0_72 (Oracle Corporation 24.72-b04)
OS:           Linux 2.6.32-042stab090.5 amd64
lein1 version
Using JLine for console I/O; install rlwrap for optimum experience.
Leiningen 1.7.1 on Java 1.7.0_72 Java HotSpot(TM) 64-Bit Server VM
lein2 version
Leiningen 2.5.0 on Java 1.7.0_72 Java HotSpot(TM) 64-Bit Server VM
lein version
Leiningen 2.5.0 on Java 1.7.0_72 Java HotSpot(TM) 64-Bit Server VM
git.checkout
0.59s$ git clone --depth=50 --branch=bump-minor-version git://github.com/mesosphere/marathon.git mesosphere/marathon
Cloning into 'mesosphere/marathon'...
remote: Counting objects: 4495, done.
remote: Compressing objects: 100% (2254/2254), done.
remote: Total 4495 (delta 2315), reused 3730 (delta 1727)
Receiving objects: 100% (4495/4495), 1.63 MiB | 0 bytes/s, done.
Resolving deltas: 100% (2315/2315), done.
Checking connectivity... done.
$ cd mesosphere/marathon
$ git checkout -qf a2862102df43243cb1733ebfc580f7ae6fa9c986
Setting environment variables from .travis.yml
$ export ARTIFACTS_AWS_REGION=us-east-1
$ export ARTIFACTS_S3_BUCKET=downloads.mesosphere.io
$ export ARTIFACTS_AWS_SECRET_ACCESS_KEY=[secure]
$ export ARTIFACTS_AWS_ACCESS_KEY_ID=[secure]
$ jdk_switcher use openjdk6
Switching to OpenJDK6 (java-1.6.0-openjdk-amd64), JAVA_HOME will be set to /usr/lib/jvm/java-6-openjdk-amd64
$ export JVM_OPTS=@/etc/sbt/jvmopts
$ export SBT_OPTS=@/etc/sbt/sbtopts
$ java -version
java version "1.6.0_33"
OpenJDK Runtime Environment (IcedTea6 1.13.5) (6b33-1.13.5-1ubuntu0.12.04)
OpenJDK 64-Bit Server VM (build 23.25-b01, mixed mode)
$ javac -version
javac 1.6.0_33
Using Scala 2.11.5
before_script.1
0.01s$ export PROJECT_VERSION="`egrep -os -e '([0-9]\.)+[0-9](-\w+)?' version.sbt`" && echo Project Version is $PROJECT_VERSION
Project Version is 0.8.0-SNAPSHOT
before_script.2
0.01s$ export ARTIFACT="marathon-$PROJECT_VERSION.tgz"
76.49s$ pushd src/main/resources/assets && npm install -g broccoli-cli && npm install && npm run dist && popd
~/build/mesosphere/marathon/src/main/resources/assets ~/build/mesosphere/marathon
--e/travis/.nvm/v0.10.33/bin/broccoli -> /home/travis/.nvm/v0.10.33/lib/node_modules/broccoli-cli/bin/broccoli
broccoli-cli@0.0.1 /home/travis/.nvm/v0.10.33/lib/node_modules/broccoli-cli
└── resolve@0.6.3
-|/\-\|-\/-|/-|/\\/\|-\ 
> cson@1.6.2 preinstall /home/travis/build/mesosphere/marathon/src/main/resources/assets/node_modules/broccoli-replace/node_modules/applause/node_modules/cson
> node ./cyclic.js
||/\|/\| 
> ambi@2.2.0 preinstall /home/travis/build/mesosphere/marathon/src/main/resources/assets/node_modules/broccoli-replace/node_modules/applause/node_modules/cson/node_modules/ambi
> node ./cyclic.js
--| 
> typechecker@2.0.8 preinstall /home/travis/build/mesosphere/marathon/src/main/resources/assets/node_modules/broccoli-replace/node_modules/applause/node_modules/cson/node_modules/extract-opts/node_modules/typechecker
> node ./cyclic.js
/| 
> typechecker@2.0.8 preinstall /home/travis/build/mesosphere/marathon/src/main/resources/assets/node_modules/broccoli-replace/node_modules/applause/node_modules/cson/node_modules/ambi/node_modules/typechecker
> node ./cyclic.js
\- 
> js2coffee@0.3.5 preinstall /home/travis/build/mesosphere/marathon/src/main/resources/assets/node_modules/broccoli-replace/node_modules/applause/node_modules/cson/node_modules/js2coffee
> node ./cyclic.js
/-|-\/-|-\/\|/\|/\|npm WARN optional dep failed, continuing fsevents@0.3.1
--\/-\|-|/\/-|-\/\|-\/\npm ERR! cb() never called!
npm ERR! not ok code 0
The command "pushd src/main/resources/assets && npm install -g broccoli-cli && npm install && npm run dist && popd" failed and exited with 1 during .
Your build has been stopped.
```

**Comment by mlunoe** [comment]
Apprently it's a bug in an old version of npm. I need to upgrade it on the travis node.

**Comment by ConnorDoyle** [comment]
@mlunoe, you should be able to pin the nodejs/npm version that the travis workers use to build the project.  See: http://docs.travis-ci.com/user/languages/javascript-with-nodejs and https://github.com/mesosphere/marathon/blob/master/.travis.yml

**Comment by mlunoe** [comment]
Hah, thanks! I'm already doing it :)

**Comment by drexin** [comment]
Thanks!

**Comment by mlunoe** [comment]
This fixed the build:
```
apt-get install -y curl
sudo apt-get install -y npm
npm install -g npm@2.0.2
npm cache clear
```

**Comment by mlunoe** [comment]
Maybe this is more stable in the future:
```
curl http://nodejs.org/dist/v0.12.0/node-v0.12.0-linux-x64.tar.gz -o node-v0.12.0-linux-x64.tar.gz
tar -xzf node-v0.12.0-linux-x64.tar.gz
export PATH=$PATH:`pwd`/node-v0.12.0-linux-x64/bin
```

**Comment by ConnorDoyle** [comment]
@mlunoe better at least make it https if you go the curl route.

**Comment by mlunoe** [comment]
:+1:

**Comment by mlunoe** [comment]
The original thread about it being fixed in `npm@2.1.3`: https://github.com/npm/npm/issues/5920#issuecomment-58600747
